### PR TITLE
Add `typos` to CI

### LINKED ISSUE #1093: Use `typos`
[`typos`](https://github.com/crate-ci/typos) currently reports two typos. Maybe add it to the CI?

### PR DESCRIPTION
**Pull request by antoniolinhart**
Hi all!

This CL fixes #1093:
* Avoids including current false-positives in the checking of typos
* Excludes localization-related files, as `typos` works with
  English words
* Fixes existing typos caught in the repo

Tested this in CI with a typo and it showed up in the list of actions!

### CODE REVIEW

**Comment by antoniolinhart** [comment]
Sorry for all of the workflow runs - was trying to see whether this worked or not. Do I need to be worried about the number of commits or does it automatically get squashed upon merging?

Thanks!!

File: _typos.toml
Comment (mgeisler) [inline _typos.toml:1]: Is it possible to "hide" this file inside the `.github/workflows/` directory? That way it doesn't take away space for everybody who browses https://github.com/google/comprehensive-rust/.

**Comment by mgeisler** [comment]
> Sorry for all of the workflow runs - was trying to see whether this worked or not.

No worries about those — I didn't get emailed about them so it was probably only spamming yourself :smile: 

> Do I need to be worried about the number of commits or does it automatically get squashed upon merging?

Because we've gotten a lot of drive-by commits, I've disabled the normal merges on this repository. We only use squash commits and the commit message defaults to the PR title and description. This makes the history much cleaner and it lowers the bar for entry for people who are not comfortable rebasing and squashing commits themselves.

So don't worry about it, just make sure the PR has a useful title and description.

**Review by mgeisler — CHANGES_REQUESTED** [review]
Thanks so much for this, it looks great! In particular, it's great that you fixed the existing typos in this PR as well.

I would love if you could move the config file inside the `.github/` directory — possibly next to the `labeler.yml` config file.

File: _typos.toml
Comment (antoniolinhart) [reply _typos.toml:1]: Yup! Moved the file and confirmed it still works properly

**Comment by antoniolinhart** [comment]
Moved the typos file and confirmed (by adding a typo) that the config file still works :+1:

**Review by mgeisler — APPROVED** [review]
Thanks, I look forward to a typo-free future :smile:

**Comment by antoniolinhart** [comment]
> Thanks, I look forward to a typo-free future 😄

Oh thanks for the fix!! I mistakenly moved it to the wrong dir

**Comment by mgeisler** [comment]
> > Thanks, I look forward to a typo-free future 😄
> 
> Oh thanks for the fix!! I mistakenly moved it to the wrong dir

All good, your testing made it easy for me to move it!

**Pull request merged**